### PR TITLE
WP Edits [WIP]

### PR DESCRIPTION
**Pull request by fdel15**
- [x] Refactors Abstract
- [ ] Refactor to white paper format from [existing template](https://www.wordlayouts.com/white-paper-template)
- [ ] Clearly Defined MVP
- [ ] Proof Read All Code snippets
- [ ] Proof Reads all Grammar/Spelling

I still don't quite get what our MVP is going to be. Here are some questions:

1. Are we creating our own marketplace that requires signing up people providing services (e.g writer) and people that require those services?

2. Are we creating a service that can be integrated into existing marketplaces (e.g. Upwork)?

3. I want to participate as a service provider (e.g. writer). What does my onboarding process look like? How do I build my trust from an unknown to trustworthy?

4. I want to participate as a judge. What does my onboarding process look like? How do I become a judge that can be trusted by two parties that do not know each other?

5. I want to participate as someone who needs to purchase a service (e.g. pay for book). What does my onboarding process look like? How can I easily find trustworthy people who can provide the service I need? How do I know I can trust the Trust Tokens? What do I need to know about judges before hand?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ reward-driven, platform for goods and service exchanges
 
 ### Supply
 
-Differently from traditional currencies, such as Bitcoin, Litecoin and Ethereum,
+<!-- -1: Ethereum is not deflationary -->
+Differently from traditional currencies, such as Bitcoin, Litecoin and Ethereum[-1],
```
Comment (fdel15) [inline README.md]: Ethereum is still minting more coins than they are burning, even after EIP-1559

https://www.theblockcrypto.com/post/113650/heres-how-much-eth-has-already-been-burned-due-to-eip-1559

**Comment by mateusap1** [comment]
1. I believe so. In my understanding we should have a "service marketplace" where people can request and offer "services", which are attached to an accusation contract (a validator script that should handle conflicts). I don't think these services should be limited to "gigs" as in other websites, though. It should be more an abstract data type whose only function is to better identify an accusation contract and set a price to it. In this sense, a company could define a set of rules to their employees and call it a "service" or a politician make some campaign promise with some tokens as a guarantee in case he doesn't follow it and also post it as a "service".
2. Yes, that should be one of our goals and that's why I think we should make DigiSevices more abstract, making it possible for people to pay services with any token they prefer, but that's up to discussion.
3. In my opinion we should take an approach similar to Uber and other similar companies, not requiring much about the user (maybe only the minimum amount of tokens) and he would build trust in the platform by having good reviews, making good deals and depositing a reasonable amount of "trust tokens" so clients feel comfortable hiring him. So to summarize everything, you would buy a membership for N amount of DSET tokens, start offering your service as a writer for a cheap amount and be more "respected" by receiving good reviews from your first clients (which weren't too afraid of hiring you since your price was cheaper).
4. Again, we didn't discuss this very much so I will give my opinion on how it could be. I believe judges shouldn't be required to do anything special other than being a member of the platform. Trust would be built in the same way it is with service providers, by having good reviews and good reputation, as well as, setting a lower price in the beginning. So, if you wanted to be a mediator, you would buy a membership, announce it in the platform and set a low percentage for your services. You could also of course be chosen by people that know you personally and trust you already.
5. Similar to the other answers, as a client you should only need to be a member of the platform in my opinion. So you would access DigiServices' website, see a service that fits your needs and make sure the selected judges are trustworthy and the contract is fair. You don't need to trust the service provider as long as you trust the judges, but you can make sure he's honest by viewing his reviews. In the same way, "trust tokens" don't need to be trusted, they are there in case there is a conflict and one of the parties is declared guilty, so if you believe the judges will provide the right inputs you don't need to worry. Judges could have a "description" section where they can provide useful information about them so that could be useful for you in order to verify their honesty. Alongside, the platform keeps a register of how many times these judges decided something that was later revoked (by asking the two next judges for input) so if a judge has too many of these, you shouldn't trust him.

**Comment by GabriIT** [comment]
I agree fully with the opinion fully aligned with the description in the WP branch I drafted adding few comments:
2.  I agree it is a good proposal to leave open the payment terms to any token or fiat currency. Yet we need to make it an incentive to use DSET to assure a successful ICO.
5. The concept is further expanded in the CAS calculation formula you can read in WP draft

**Comment by mateusap1** [comment]
We should change this function

```haskell
-- The total amount of tokens that will be minted every month
totalAmt :: Int
totalAmt = 1000

-- Review is an integer between 0 and 100
calculateRewards :: [Review] -> [Int]
calculateRewards [] = []
calculateRewards (x:xs) = ((x `div` revSum) * totalAmt `div` 100) : calculateRewards xs
  where
    revSum :: Integer
    revSum = sum xs
```

to the new one I made

```haskell
type CAS = Integer

-- The total amount of tokens that will be minted every month
totalAmt :: Integer
totalAmt = 1000

calculateReward :: CAS -> [CAS] -> Integer
calculateReward score xs = (score `timesInteger` totalAmt) `div` sum xs

calculateRewards :: [CAS] -> [Integer]
calculateRewards xs = foldr (\x acc -> calculateReward x xs : acc) [] xs
```

**Comment by fdel15** [comment]
> ```haskell
> 
> calculateReward :: CAS -> [CAS] -> Integer
> calculateReward score xs = (score `timesInteger` totalAmt) `div` sum xs
> ```

Before the comment said `Review` could be a number between 0 and 100. Is it possible that the sum will be 0? If so, we need to handle this scenario because I think dividing by 0 will cause a runtime error.

**Comment by mateusap1** [comment]
@fdel15 is right, a simple "guard" will do the trick I think

**Comment by mateusap1** [comment]
What about this?
```haskell
type CAS = Integer

-- The total amount of tokens that will be minted every month
totalAmt :: Integer
totalAmt = 1000

calculateReward :: CAS -> [CAS] -> Integer
calculateReward score xs
  | sum xs == 0 = 0
  | otherwise = (score * totalAmt) `div` sum xs

calculateRewards :: [CAS] -> [Integer]
calculateRewards xs = foldr (\x acc -> calculateReward x xs : acc) [] xs
```

**Comment by GabriIT** [comment]
@mateusap1 ok

**Comment by GabriIT** [comment]
an alternative: https://www.slideserve.com/enrico/an-introduction-to-monads

**Comment by mateusap1** [comment]
I replaced `timesInteger` with `*` since it was not necessary and would require us to import another module

**Pull request merged**